From: Sebastian Rasmussen <sebras@gmail.com>
Date: Wed, 26 Sep 2018 03:31:05 +0800
Subject: [PATCH] Throw when page number is out of range.

Other code depends on this and does not handle
a page pointer being NULL well.

Origin: https://bugs.ghostscript.com/show_bug.cgi?id=700446
---
 source/cbz/mucbz.c   | 2 +-
 source/cbz/muimg.c   | 2 +-
 source/svg/svg-doc.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/source/cbz/mucbz.c b/source/cbz/mucbz.c
index 5cbdf0c..3aef043 100644
--- a/source/cbz/mucbz.c
+++ b/source/cbz/mucbz.c
@@ -181,7 +181,7 @@ cbz_load_page(fz_context *ctx, fz_document *doc_, int number)
 	fz_buffer *buf = NULL;
 
 	if (number < 0 || number >= doc->page_count)
-		return NULL;
+		fz_throw(ctx, FZ_ERROR_GENERIC, "cannot load page %d", number);
 
 	fz_var(page);
 
diff --git a/source/cbz/muimg.c b/source/cbz/muimg.c
index b54371f..fb07530 100644
--- a/source/cbz/muimg.c
+++ b/source/cbz/muimg.c
@@ -82,7 +82,7 @@ img_load_page(fz_context *ctx, fz_document *doc_, int number)
 	img_page *page = NULL;
 
 	if (number < 0 || number >= doc->page_count)
-		return NULL;
+		fz_throw(ctx, FZ_ERROR_GENERIC, "cannot load page %d", number);
 
 	fz_var(pixmap);
 	fz_var(image);
diff --git a/source/svg/svg-doc.c b/source/svg/svg-doc.c
index 112368c..8ac20cb 100644
--- a/source/svg/svg-doc.c
+++ b/source/svg/svg-doc.c
@@ -55,7 +55,7 @@ svg_load_page(fz_context *ctx, fz_document *doc_, int number)
 	svg_page *page;
 
 	if (number != 0)
-		return NULL;
+		fz_throw(ctx, FZ_ERROR_GENERIC, "cannot find page %d", number);
 
 	page = fz_new_derived_page(ctx, svg_page);
 	page->super.bound_page = svg_bound_page;
